### PR TITLE
ARTEMIS-1581 fix handshake-timeout property configurability

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

**Comment by jbertram** [comment]
GitHub is reporting that this branch has conflicts.  Also it would be worth adding a test here to verify the fix since none of the existing tests caught it.

**Comment by stanlyDoge** [comment]
The build failed at MessageConsumerTest.testStopConnectionDuringOnMessage test, but it passes at local. Is that just coincidence it has failed? Can we re-run jenkins build?

**Comment by michaelandrepearce** [comment]
@stanlyDoge could you squash the commits. Cheers.

**Comment by stanlyDoge** [comment]
@michaelandrepearce Done
@jbertram ahh, that explains my another conflict. Thanks.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java [hunk 1]
```diff
@@ -97,4 +111,56 @@ public void testSetupJMSConfiguration() throws Exception {
 
       server.stop();
    }
+
+   @Test
+   public void testHandshakeTimeoutWithValueSet() throws Exception {
+      JMSServerManager jmsServer;
```
Comment (jbertram) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java]: The JMSServerManager is deprecated so it's probably best not to use it in new tests.  Check out the "embedded-simple" example for how to programmatically instantiate a broker with a configuration file.

Comment (jbertram) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java]: Or don't even use a configuration file and just programmatically configure the acceptor.

Comment (stanlyDoge) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java]: Good point with JMSServerManager, thanks. 
The issue is about unability to set timeout value using conf file so using it is intentional. Setting this value programmatically is done [here](https://github.com/apache/activemq-artemis/pull/1534/files#diff-70ef2b6fc0b6f5e37ea6023acce65b8c). Or did you think anything else?

Comment (jbertram) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java]: The problematic use-case wasn't necessarily with a configuration file, but simply using a URL (which can be done programmatically).  I changed the test a bit before I merged it.  It's a bit simpler this way.

Comment (stanlyDoge) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/server/config/JMSConfigurationTest.java]: Aha. Thank you :)
